metricbeat/module/ceph/osd_tree/osd_tree.go

```diff
@@ -63,24 +67,20 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	content, err := m.HTTP.FetchContent()
 	if err != nil {
-		m.Logger().Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "error in fetch")
 	}
 
 	events, err := eventsMapping(content)
 	if err != nil {
-		m.Logger().Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "error in mapping")
 	}
 
 	for _, event := range events {
 		reporter.Event(mb.Event{MetricSetFields: event})
```

Comment:
Add `if !reported { return nil }` check to be consistent with https://github.com/elastic/beats/pull/11979/files#diff-dc97d499b303003462f444c7164c2780R78?